programs/programs/ma/cha/tests.py

```diff
@@ -0,0 +1,226 @@
+from django.test import TestCase
+from programs.programs.ma.cha.calculator import Cha
+from screener.models import Screen, HouseholdMember, IncomeStream, WhiteLabel
+from programs.models import Program, FederalPoveryLimit
+from programs.util import Dependencies
+from unittest.mock import patch, MagicMock
+
+
+class TestCha(TestCase):
+    """Test cases for Cambridge Housing Authority calculator"""
+
+    @classmethod
+    def setUpTestData(cls):
+        """Set up test data that doesn't change between tests"""
+        cls.ma_white_label = WhiteLabel.objects.create(name="Massachusetts", code="ma", state_code="MA")
+        cls.fpl_year = FederalPoveryLimit.objects.create(year="2025", period="2025")
+        cls.program = Program.objects.new_program(white_label="ma", name_abbreviated="ma_cha")
+        cls.program.year = cls.fpl_year
+        cls.program.save()
+
+    def setUp(self):
+        """Set up test fixtures for each test method"""
+        self.eligible_screen = Screen.objects.create(
+            agree_to_tos=True,
+            zipcode="02138",
+            county="Cambridge",
+            household_size=2,
+            white_label=self.ma_white_label,
+            completed=False,
+        )
+
+        self.head = HouseholdMember.objects.create(
+            screen=self.eligible_screen,
+            relationship="headOfHousehold",
+            age=35,
+            student=False,
+            has_income=True,
+        )
+
+        # Income below 50% AMI (approx $50,000/year for 2-person household)
+        IncomeStream.objects.create(
+            screen=self.eligible_screen,
+            household_member=self.head,
+            type="wages",
+            amount=3000,  # $36,000/year
+            frequency="monthly",
+        )
+
+    def create_calculator(self, screen):
+        """Helper method to create calculator instance"""
+        data = {}
+        missing_dependencies = Dependencies()
+        return Cha(screen, self.program, data, missing_dependencies)
+
+    @patch("programs.programs.ma.cha.calculator.hud_client")
+    def test_household_eligible_in_cambridge_below_income_limit(self, mock_hud_client):
+        """Test household is eligible when in Cambridge and below 50% AMI"""
+        mock_hud_client.get_screen_il_ami.return_value = 50000  # 50% AMI limit
+
+        calc = self.create_calculator(self.eligible_screen)
+        eligibility = calc.eligible()
+
+        self.assertTrue(eligibility.eligible)
+        mock_hud_client.get_screen_il_ami.assert_called_once_with(
+            self.eligible_screen, "80%", "2025", county_override="Middlesex"
```

Comment:
I'm having trouble following the 50% comments above (lines 40, 57, 58) and the 80% expectation. Are the comments accurate and I'm missing something? Or should they say below 80%? Looks like some of the other tests say 80%.

Reply:
Ahh, good catch! I thought I caught all of these, but apparently I did not. (For context: my initial implementation used 50% before I realized that other benefits from this program are available up to 80%)

Will fix these comments!